File: WOMtest.js
// WOMtest.js
// A test script to gather data from the Wise Old Man API, now with javascript!

const { app, BrowserWindow, ipcMain } = require('electron');
const { WOMClient } = require('@wise-old-man/utils');
const fs = require('fs');
const path = require('path');

let mainWindow;

function createWindow() {
    mainWindow = new BrowserWindow({
        width: 800,
        height: 600,
        webPreferences: {
            nodeIntegration: true,
            contextIsolation: false
        }
    });

    mainWindow.loadFile('index.html');
}

// Helper function to log messages with timestamp
function log(message) {
    const timestamp = new Date().toISOString();
    const logMessage = `${timestamp} - ${message}`;
    console.log(logMessage);
    if (mainWindow) {
        mainWindow.webContents.send('log', logMessage);
    }
}

async function fetchGroupDetails(groupId) {
    try {
        // Initialize WOM client
        const client = new WOMClient();
        log("WOM client initialized");

        // Fetch group details
        const result = await client.groups.getGroupDetails(groupId);
        log("Group details fetched successfully");

        // Create output directory if it doesn't exist
        const outputDir = path.join(__dirname, 'output');
        if (!fs.existsSync(outputDir)) {
            fs.mkdirSync(outputDir);
        }

        // Save to JSON file with pretty formatting
        const outputPath = path.join(outputDir, 'group_details.json');
        fs.writeFileSync(
            outputPath,
            JSON.stringify(result, null, 2)
        );
        log(`Group details saved to ${outputPath}`);

        return {
            name: result.name,
            memberCount: result.memberCount,
            createdAt: result.createdAt,
            updatedAt: result.updatedAt
        };

    } catch (error) {
        log(`Error: ${error.message}`);
        throw error;
    }
}

// Set up electron app
app.whenReady().then(createWindow);

app.on('window-all-closed', () => {
    if (process.platform !== 'darwin') {
        app.quit();
    }
});

app.on('activate', () => {
    if (BrowserWindow.getAllWindows().length === 0) {
        createWindow();
    }
});

// Handle IPC messages from renderer
ipcMain.handle('fetch-group', async (event, groupId) => {
    try {
        return await fetchGroupDetails(groupId);
    } catch (error) {
        return { error: error.message };
    }
});

File: WOMtest.py
# WOMtest.py
# A test script to gather data from the Wise Old Man API

from wom import Client
import configparser
import os
from datetime import datetime
import asyncio
from discord.ext import tasks
import sys
# ------------------------------------------------------------------------------
# Helper Functions
# ------------------------------------------------------------------------------

def setup_logging():
    """Sets up logging to both file and console"""
    log_file = open("log.txt", "w")
    
    class TeeOutput:
        def __init__(self, filename, original_stdout):
            self.terminal = original_stdout
            self.log_file = open(filename, "w")
        
        def write(self, message):
            self.terminal.write(message)
            self.log_file.write(message)
        
        def flush(self):
            self.terminal.flush()
            self.log_file.flush()
        
        def close(self):
            self.log_file.close()
    
    sys.stdout = TeeOutput("log.txt", sys.stdout)

def log(message: str):
    """Logs a message with the current timestamp."""
    timestamp = datetime.now().strftime("%Y-%m-%d %H:%M:%S")
    formatted_message = f"{timestamp} - {message}"
    print(formatted_message)  # Print to terminal

# ------------------------------------------------------------------------------
# Configuration Loading
# ------------------------------------------------------------------------------

config = configparser.ConfigParser()
config_file = os.path.join(os.path.dirname(__file__), 'config.ini')
config.read(config_file)

# Fetch settings from config.ini	
channel_id          = int(config['discord']['channel_id'])
group_id            = int(config['wiseoldman']['group_id'])
group_passcode      = config['wiseoldman']['group_passcode']

# ------------------------------------------------------------------------------
# Wise Old Man Client Initialization
# ------------------------------------------------------------------------------

wom_client = Client()

async def main():
    # Setup logging
    setup_logging()
    
    # Start the WOM client session
    await wom_client.start()
    log("WOM client started")
    
    # Fetch group details
    result = await wom_client.groups.get_details(group_id)
    log("Group details fetched successfully")
    print("\nGroup details:")
    print(result)
    
    # Close the client session when done
    await wom_client.close()
    log("WOM client closed")
    
    # Restore original stdout and close log file
    sys.stdout.close()
    sys.stdout = sys.__stdout__

asyncio.run(main())











